### PR TITLE
[testing cronjob] add scripts for adding a testing cronjob

### PR DESCRIPTION
**Pull request by chejennifer**
- script for pushing the docker image
- script for adding a periodic testing cronjob to an environment
- updates to website deployment to be able to run this cronjob

### CODE REVIEW

File: build/website_cron_testing/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Website Cron Testing Docker Image
+
+## Description
+
+This is a Docker image based on python:3.11, but comes with:
+
+- Google Chrome: the browser which is used to run the tests.
+- ChromeDriver: used to send commands to Google Chrome.
+- Script (and files required by the script) to run the website cron tests
+
```
Comment (shifucun) [inline build/website_cron_testing/README.md:10]: If you are able to use gcr.io/datcom-ci/webdriver-chrome:2023-06-17, can update here as well

File: build/website_cron_testing/run_website_cron_tests.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+echo "Website domain: $WEB_API_ROOT"
+echo "Nodejs domain: $NODEJS_API_ROOT"
+echo "Testing env: $TESTING_ENV"
+
+NO_PIP=$2
+
+python3 -m venv .env
+source .env/bin/activate
+if [[ $NO_PIP != "true" ]]; then
+  python3 -m pip install --upgrade pip setuptools
```
[the remark this answers is not in the document]
Comment (shifucun) [reply build/website_cron_testing/run_website_cron_tests.sh:29]: If the requirements.txt is from other folder, you can make a symbolic link to server/requirements.txt in this folder

File: gke/cron_testing_job.yaml.tpl [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Copyright 2020 Google LLC
```
Comment (shifucun) [inline gke/cron_testing_job.yaml.tpl:1]: 2023

**Review by shifucun — COMMENTED** [review]
This is very cool! Most comments about the docker image.

File: build/website_cron_testing/run_website_cron_tests.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (chejennifer) [reply build/website_cron_testing/run_website_cron_tests.sh:29]: this script is going to be used to run all the tests. I'm going to add to this script as I add the tests.

But I can move pip install to the dockerfile

[the remark this answers is not in the document]
Comment (chejennifer) [reply build/website_cron_testing/run_website_cron_tests.sh:29]: sorry, do you mean keep the pip install in this script or still move it to the dockerfile?

[the remark this answers is not in the document]
Comment (shifucun) [reply build/website_cron_testing/run_website_cron_tests.sh]: Move the pip install to Dockerfile.

File: build/website_cron_testing/README.md (hunk 1 above)
Comment (chejennifer) [reply build/website_cron_testing/README.md]: done

File: build/website_cron_testing/run_website_cron_tests.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (chejennifer) [reply build/website_cron_testing/run_website_cron_tests.sh]: done

File: gke/cron_testing_job.yaml.tpl [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# limitations under the License.
+
+
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: cron-testing
+  namespace: website
+spec:
+  schedule: "0 */4 * * *"
```
Comment (shifucun) [inline gke/cron_testing_job.yaml.tpl:22]: Add a comment on the actual schedule here, is this every 4 hours?

**Review by shifucun — APPROVED** [review]
very nice!

File: gke/cron_testing_job.yaml.tpl (hunk 4 above)
Comment (chejennifer) [reply gke/cron_testing_job.yaml.tpl:23]: yes, every 4 hours & added a comment

File: gke/README.md [hunk 5]
```diff
@@ -126,4 +126,15 @@ gcloud config set project <PROJECT>
...
-where `<ENV>` refers to the name of the instance and `<REGION>` is the region of the cluster.
\ No newline at end of file
+where `<ENV>` refers to the name of the instance and `<REGION>` is the region of the cluster.
+
+## Add a periodic testing job
+
+To add a cronjob to run periodic testing against a cluster, run:
+
+```bash
+./setup_periodic_testing.sh -e <ENV> -l <REGION>
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply gke/README.md:136]: Just need to run once to set up the job. The job uses the latest docker image, so no need to re-run the script when docker image is updated

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**